LandonHotel.Tests/BookingServiceTests.cs

```diff
@@ -71,11 +71,15 @@ public void IsBookingCostForAdditionalPersonsCorrect()
             //TODO: Calculate total booking cost using the constant values and the business rule
             //           cost per day = rate + rate * guestCount * 10/100
 
-            var totalBookingCost = 1.0; //dummy value to start
+            var totalBookingCost = (rate + (rate * (guestCount-1) * 10/100)) * numberDays; //first guest rate, and add rate for (guests excluding the first guest) at 10% increase, multiplied by number of days
 
             //Act
             //TODO: Setup roomRepo and create new Booking using the constants
-            Booking booking = null;
+            Booking booking = new Booking();
+            booking.CheckInDate = checkin;
+            booking.CheckInDate = checkout;
```

Comment:
Use "booking.CheckOutDate" instead
This is why you are getting the null reference exception in your service

Reply:
CheckInDate changed to CheckOutDate. Hard Coded values all removed. RoomRepo setup but still isn't working for me